pallets/confidential/src/rng.rs

```diff
@@ -0,0 +1,118 @@
+use rand_core::{CryptoRng, Error, RngCore};
+use sp_runtime_interface::runtime_interface;
+
+#[cfg(not(feature = "std"))]
+use core::num::NonZeroU32;
+
+#[cfg(feature = "std")]
+use rand::rngs::OsRng;
+
+/// Host functions to allow native access to OS RNG from Wasm.
+///
+/// # Change Log
+/// - Runtime Spec version 1005. It is incompatible with previous versions, and it requires a
+/// binary update.
+///
+/// # TODO
+///  - Use lazy static RNG object to mitigate performance impact during the `OsRng` object
+///  creation. We should double-check the security implications of having a global RNG in a
```

Comment:
This is a very good question. On Linux, getting a random number involves making a system call to [`getrandom`](https://github.com/torvalds/linux/blob/bfe91da29bfad9941d5d703d45e29f0812a20724/drivers/char/random.c#L2007) which accesses `/dev/urandom`. Every user thread calling `getrandom` will get a corresponding kernel thread by the way Linux syscalls work. Hence the calls themselves can be concurrent. But since `/dev/urandom` is a streaming character device, the random data received by different threads is going to be also different and independent unless they read it at the same exact time which I think should be impossible because of the sequential access to the file system.

Having a global `OsRng` in Linux looks fine.

[Other OS](https://docs.rs/getrandom/0.1.14/getrandom/) can be also checked.

Reply:
That concurrency problem is solved by `ThreadRng`, and that generates new complexities into the game.
I did a simple and **no efficient** solution because `cryptography` team is _re-evaluating the use of `RNG` as part of its interface_. They want to replace it by the hash of some inputs with multiple chained challenges. So if that works, we could remove the use of RNG completely. 

`getrandom` crate does not work in our context. We'll see that in the knowledge-transfer process :wink:

Reply:
OK with a remark that `OsRng` actually uses `getrandom`.